src/projects/TodoList/index.js

```diff
@@ -43,20 +46,45 @@ class TodoList extends React.Component {
     // taskListToBeAltered.splice(i, 1, alteredTask);
     this.setState({ todos: taskListToBeAltered });
   }
+  showEditField(i) {
+    let taskListToBeAltered = this.state.todos;
+    let alteredTask = taskListToBeAltered[i];
+    alteredTask.isEditing = !alteredTask.isEditing;
+    this.setState({ todos: taskListToBeAltered });
+    this.setState({ editedItem: alteredTask.task });
+  }
+  confirmEditTask(i) {
+    let taskListToBeAltered = this.state.todos;
+    let alteredTask = taskListToBeAltered[i];
+    alteredTask.task = this.state.editedItem;
+    this.setState({ todos: taskListToBeAltered });
+    alteredTask.isEditing = !alteredTask.isEditing;
+  }
+
   render() {
     // Before the return you can do all logic you need
     let todos = this.state.todos.map((task, i) => {
       return (
         <Todo
           key={i}
-          task={task}
+          todo={task.task}
+          completed={task.completed}
+          isEditing={task.isEditing}
+          editedItem={this.state.editedItem}
```

Comment:
This here is the crux of the bug you mentioned in your main note - and your suggestion for how to fix is bang on. Each todo task should have its own "edited" value which is passed in to the edit input and changed, then you use that value to overwrite the main `todo.task` when they hit save, or you clear it out if they hit cancel.

Doing it this way would work if you could only edit a single todo at a time (remember, top level keys in state are 'global' to the component and the array of objects for todos contain todo specific things)

Reply:
right, will get on that and sort it. cant imagine it'll take too long